pkg/pillar/types/assignableadapters.go

```diff
@@ -464,24 +462,39 @@ func (aa *AssignableAdapters) LookupIoBundleIfName(ifname string) *IoBundle {
 func (aa *AssignableAdapters) CheckParentAssigngrp() bool {
 	assigngrp2parent := make(map[string]string)
 
+	ownParentError := "IOBundle cannot be it's own parent"
+	parentAssigngrpMismatchError := "IOBundle with parentassigngrp mismatch found"
+	emptyAssigngrpWithParentError := "IOBundle with empty assigngrp cannot have a parent"
+	cycleDetectedError := "Cycle detected, please check provided parentassigngrp/assigngrp"
+
 	var cycleDetectedAssigngrp string
 	for i := range aa.IoBundleList {
 		ioBundle := &aa.IoBundleList[i]
 
+		for _, errStr := range []string{
+			ownParentError,
+			parentAssigngrpMismatchError,
+			emptyAssigngrpWithParentError,
+			cycleDetectedError,
+		} {
+			if ioBundle.Error == errStr {
```

Comment:
Are we sure the error messages are not modified in between, and can we expect them to be the same as we set them in the function? Sometimes, we add prefixes/postfixes to error messages...

Reply:
Oh, I see... I was reviewing per commit. Will you rework the commits?

Reply:
You want the commits to be squashed?
For me both is okay, this one does eliminate unnecessary commits, the other way the stable branch and master branch have the same commit (I don't think we should backport this to stable).

Reply:
Now they're squashed.